src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryBasicInfo.java

```diff
@@ -10,12 +17,87 @@
   static final String STAGE_NAME = "stage_name";
   static final String PIPE_NAME = "pipe_name";
 
+  // Name of mBean if jmx monitoring is enabled
+  private ObjectName mBeanName;
+
+  // A boolean to turn on or off a JMX metric as required.
+  private volatile boolean enableJMXMonitoring;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeTelemetryBasicInfo.class);
+
+  private static final String JMX_METRIC_PREFIX = "snowflake.kafka.connector";
+
   SnowflakeTelemetryBasicInfo(
-      final String tableName, final String stageName, final String pipeName) {
+      final String tableName,
+      final String stageName,
+      final String pipeName,
+      final String connectorName,
+      final boolean enableJMXMonitoring) {
     this.tableName = tableName;
     this.stageName = stageName;
     this.pipeName = pipeName;
+    this.enableJMXMonitoring = enableJMXMonitoring;
+    if (enableJMXMonitoring) {
+      mBeanName = metricName(pipeName, connectorName);
+    }
   }
 
   abstract void dumpTo(ObjectNode msg);
+
+  /**
+   * Registers a metrics MBean into the platform MBean server. The method is intentionally
+   * synchronized to prevent preemption between registration and unregistration.
+   */
+  public synchronized void registerMBean() {
+    try {
+      final MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+      if (mBeanServer == null) {
+        LOGGER.info(Logging.logMessage("JMX not supported, bean '{}' not registered", mBeanName));
+        return;
+      }
+      mBeanServer.registerMBean(this, mBeanName);
+      LOGGER.info(Logging.logMessage("Registered Mbean:{}", mBeanName));
+    } catch (JMException e) {
```

Comment:
Thinking out loudly here, is it okay to fail the sinkService start if any other exception is thrown? Should we have a catch all exception here for safety?

Reply:
Yes, makes sense to catch all. Also just realised `ManagementFactory.getPlatformMBeanServer` throws SecurityException (Extends RTE) so it might be worth to catch all exceptions. I will do that instead of JMException